src/include/btree_inline.h

```diff
@@ -2344,6 +2336,7 @@ __wt_btcur_skip_page(
      * if the page has been modified since it was reconciled, since neither the delete information
      * nor the timestamp information is necessarily up to date.
      */
+    WT_ASSERT(session, __wt_gen(session, WT_GEN_SPLIT) != 0);
```

Comment:
Could this be moved to a WT_ASSERT_ALWAYS within the __wt_ref_addr_copy implementation? It might need to go after the check that the addr is non-nul.

Reply:
The `__wt_ref_addr_copy` is called in locations where the split generation is not taken, so moving the assert can fail. This is the only location where we can get into problems, so placing the assert here would be good.